File: adapter-pattern/js/src/AudioPlayer.js
import { MediaPlayer } from './MediaPlayer.js'

// 音频播放器
export class AudioPlayer extends MediaPlayer {
  // constructor(fileType, filePath) {
  //   super(fileType, filePath)
  // }

  play() {
    console.log(
      'AudioPlayer fileType:' + this.fileType + ' | filePath:' + this.filePath,
    )
    console.log('AudioPlayer::play()')
    if (this.fileType === 'mp3') {
      console.log('AudioPlayer::play() mp3')
    } else if (this.fileType === 'mp4') {
      console.log('AudioPlayer::play() mp4')
    }
  }
}

File: adapter-pattern/go/src/AudioPlayer.go
package src

import (
  "fmt"
)

// 音频播放器
type AudioPlayer struct {
  FileType string
  FilePath string
}

func (a *AudioPlayer) SetFileType(fileType string) {
  a.FileType = fileType
}

func (a *AudioPlayer) SetFilePath(filePath string) {
  a.FilePath = filePath
}

func (a *AudioPlayer) Play() {
  fmt.Println("AudioPlayer FileType:" + a.FileType + " | FilePath:" + a.FilePath)
  fmt.Println("AudioPlayer::Play()")
  if a.FileType == "mp3" {
    fmt.Println("AudioPlayer::Play() mp3")
  } else if a.FileType == "mp4" {
    fmt.Println("AudioPlayer::Play() mp4")
  }
}
